[2024-10-21T22:08:45.928+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-10-21T22:08:45.934+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: newsapi_etl_with_spark.spark_submit_task manual__2024-10-21T22:08:38.616113+00:00 [queued]>
[2024-10-21T22:08:45.938+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: newsapi_etl_with_spark.spark_submit_task manual__2024-10-21T22:08:38.616113+00:00 [queued]>
[2024-10-21T22:08:45.938+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 2
[2024-10-21T22:08:45.945+0000] {taskinstance.py:2888} INFO - Executing <Task(SparkSubmitOperator): spark_submit_task> on 2024-10-21 22:08:38.616113+00:00
[2024-10-21T22:08:45.946+0000] {standard_task_runner.py:72} INFO - Started process 241 to run task
[2024-10-21T22:08:45.948+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'newsapi_etl_with_spark', 'spark_submit_task', 'manual__2024-10-21T22:08:38.616113+00:00', '--job-id', '33', '--raw', '--subdir', 'DAGS_FOLDER/newsapi_etl_pipeline.py', '--cfg-path', '/tmp/tmpj7k7qdpb']
[2024-10-21T22:08:45.949+0000] {standard_task_runner.py:105} INFO - Job 33: Subtask spark_submit_task
[2024-10-21T22:08:45.969+0000] {task_command.py:467} INFO - Running <TaskInstance: newsapi_etl_with_spark.spark_submit_task manual__2024-10-21T22:08:38.616113+00:00 [running]> on host 17bb0edcf65f
[2024-10-21T22:08:46.022+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='newsapi_etl_with_spark' AIRFLOW_CTX_TASK_ID='spark_submit_task' AIRFLOW_CTX_EXECUTION_DATE='2024-10-21T22:08:38.616113+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-10-21T22:08:38.616113+00:00'
[2024-10-21T22:08:46.022+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-10-21T22:08:46.027+0000] {base.py:84} INFO - Retrieving connection 'spark_conn'
[2024-10-21T22:08:46.028+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --conf spark.ui.port=4050 --conf spark.executor.instances=3 --conf spark.executor.memory=5g --conf spark.executor.cores=2 --conf spark.driver.memory=4g --conf spark.driver.cores=2 --conf spark.default.parallelism=12 --conf spark.serializer=org.apache.spark.serializer.KryoSerializer --conf spark.sql.shuffle.partitions=12 --conf spark.dynamicAllocation.enabled=true --conf spark.dynamicAllocation.minExecutors=3 --conf spark.dynamicAllocation.maxExecutors=6 --conf spark.executor.memoryOverhead=1024m --name LDA_Job --verbose --deploy-mode client /opt/***/scripts/news_processing.py
[2024-10-21T22:08:46.588+0000] {spark_submit.py:649} INFO - Using properties file: null
[2024-10-21T22:08:46.641+0000] {spark_submit.py:649} INFO - Parsed arguments:
[2024-10-21T22:08:46.641+0000] {spark_submit.py:649} INFO - master                  spark://spark-master:7077
[2024-10-21T22:08:46.642+0000] {spark_submit.py:649} INFO - remote                  null
[2024-10-21T22:08:46.642+0000] {spark_submit.py:649} INFO - deployMode              client
[2024-10-21T22:08:46.642+0000] {spark_submit.py:649} INFO - executorMemory          5g
[2024-10-21T22:08:46.642+0000] {spark_submit.py:649} INFO - executorCores           2
[2024-10-21T22:08:46.642+0000] {spark_submit.py:649} INFO - totalExecutorCores      null
[2024-10-21T22:08:46.643+0000] {spark_submit.py:649} INFO - propertiesFile          null
[2024-10-21T22:08:46.643+0000] {spark_submit.py:649} INFO - driverMemory            4g
[2024-10-21T22:08:46.643+0000] {spark_submit.py:649} INFO - driverCores             2
[2024-10-21T22:08:46.643+0000] {spark_submit.py:649} INFO - driverExtraClassPath    null
[2024-10-21T22:08:46.643+0000] {spark_submit.py:649} INFO - driverExtraLibraryPath  null
[2024-10-21T22:08:46.643+0000] {spark_submit.py:649} INFO - driverExtraJavaOptions  null
[2024-10-21T22:08:46.644+0000] {spark_submit.py:649} INFO - supervise               false
[2024-10-21T22:08:46.644+0000] {spark_submit.py:649} INFO - queue                   null
[2024-10-21T22:08:46.644+0000] {spark_submit.py:649} INFO - numExecutors            3
[2024-10-21T22:08:46.644+0000] {spark_submit.py:649} INFO - files                   null
[2024-10-21T22:08:46.644+0000] {spark_submit.py:649} INFO - pyFiles                 null
[2024-10-21T22:08:46.645+0000] {spark_submit.py:649} INFO - archives                null
[2024-10-21T22:08:46.645+0000] {spark_submit.py:649} INFO - mainClass               null
[2024-10-21T22:08:46.645+0000] {spark_submit.py:649} INFO - primaryResource         file:/opt/***/scripts/news_processing.py
[2024-10-21T22:08:46.645+0000] {spark_submit.py:649} INFO - name                    LDA_Job
[2024-10-21T22:08:46.645+0000] {spark_submit.py:649} INFO - childArgs               []
[2024-10-21T22:08:46.646+0000] {spark_submit.py:649} INFO - jars                    null
[2024-10-21T22:08:46.646+0000] {spark_submit.py:649} INFO - packages                null
[2024-10-21T22:08:46.646+0000] {spark_submit.py:649} INFO - packagesExclusions      null
[2024-10-21T22:08:46.646+0000] {spark_submit.py:649} INFO - repositories            null
[2024-10-21T22:08:46.646+0000] {spark_submit.py:649} INFO - verbose                 true
[2024-10-21T22:08:46.646+0000] {spark_submit.py:649} INFO - 
[2024-10-21T22:08:46.647+0000] {spark_submit.py:649} INFO - Spark properties used, including those specified through
[2024-10-21T22:08:46.647+0000] {spark_submit.py:649} INFO - --conf and those from the properties file null:
[2024-10-21T22:08:46.647+0000] {spark_submit.py:649} INFO - (spark.default.parallelism,12)
[2024-10-21T22:08:46.647+0000] {spark_submit.py:649} INFO - (spark.driver.cores,2)
[2024-10-21T22:08:46.647+0000] {spark_submit.py:649} INFO - (spark.driver.memory,4g)
[2024-10-21T22:08:46.647+0000] {spark_submit.py:649} INFO - (spark.dynamicAllocation.enabled,true)
[2024-10-21T22:08:46.647+0000] {spark_submit.py:649} INFO - (spark.dynamicAllocation.maxExecutors,6)
[2024-10-21T22:08:46.648+0000] {spark_submit.py:649} INFO - (spark.dynamicAllocation.minExecutors,3)
[2024-10-21T22:08:46.648+0000] {spark_submit.py:649} INFO - (spark.executor.cores,2)
[2024-10-21T22:08:46.648+0000] {spark_submit.py:649} INFO - (spark.executor.instances,3)
[2024-10-21T22:08:46.648+0000] {spark_submit.py:649} INFO - (spark.executor.memory,5g)
[2024-10-21T22:08:46.648+0000] {spark_submit.py:649} INFO - (spark.executor.memoryOverhead,1024m)
[2024-10-21T22:08:46.648+0000] {spark_submit.py:649} INFO - (spark.serializer,org.apache.spark.serializer.KryoSerializer)
[2024-10-21T22:08:46.648+0000] {spark_submit.py:649} INFO - (spark.sql.shuffle.partitions,12)
[2024-10-21T22:08:46.649+0000] {spark_submit.py:649} INFO - (spark.ui.port,4050)
[2024-10-21T22:08:46.649+0000] {spark_submit.py:649} INFO - 
[2024-10-21T22:08:46.649+0000] {spark_submit.py:649} INFO - 
[2024-10-21T22:08:46.758+0000] {spark_submit.py:649} INFO - Main class:
[2024-10-21T22:08:46.758+0000] {spark_submit.py:649} INFO - org.apache.spark.deploy.PythonRunner
[2024-10-21T22:08:46.758+0000] {spark_submit.py:649} INFO - Arguments:
[2024-10-21T22:08:46.759+0000] {spark_submit.py:649} INFO - file:/opt/***/scripts/news_processing.py
[2024-10-21T22:08:46.759+0000] {spark_submit.py:649} INFO - null
[2024-10-21T22:08:46.759+0000] {spark_submit.py:649} INFO - Spark config:
[2024-10-21T22:08:46.759+0000] {spark_submit.py:649} INFO - (spark.app.name,LDA_Job)
[2024-10-21T22:08:46.760+0000] {spark_submit.py:649} INFO - (spark.app.submitTime,1729548526748)
[2024-10-21T22:08:46.760+0000] {spark_submit.py:649} INFO - (spark.default.parallelism,12)
[2024-10-21T22:08:46.760+0000] {spark_submit.py:649} INFO - (spark.driver.cores,2)
[2024-10-21T22:08:46.760+0000] {spark_submit.py:649} INFO - (spark.driver.memory,4g)
[2024-10-21T22:08:46.760+0000] {spark_submit.py:649} INFO - (spark.dynamicAllocation.enabled,true)
[2024-10-21T22:08:46.761+0000] {spark_submit.py:649} INFO - (spark.dynamicAllocation.maxExecutors,6)
[2024-10-21T22:08:46.761+0000] {spark_submit.py:649} INFO - (spark.dynamicAllocation.minExecutors,3)
[2024-10-21T22:08:46.761+0000] {spark_submit.py:649} INFO - (spark.executor.cores,2)
[2024-10-21T22:08:46.761+0000] {spark_submit.py:649} INFO - (spark.executor.instances,3)
[2024-10-21T22:08:46.762+0000] {spark_submit.py:649} INFO - (spark.executor.memory,5g)
[2024-10-21T22:08:46.762+0000] {spark_submit.py:649} INFO - (spark.executor.memoryOverhead,1024m)
[2024-10-21T22:08:46.762+0000] {spark_submit.py:649} INFO - (spark.master,spark://spark-master:7077)
[2024-10-21T22:08:46.762+0000] {spark_submit.py:649} INFO - (spark.serializer,org.apache.spark.serializer.KryoSerializer)
[2024-10-21T22:08:46.762+0000] {spark_submit.py:649} INFO - (spark.sql.shuffle.partitions,12)
[2024-10-21T22:08:46.763+0000] {spark_submit.py:649} INFO - (spark.submit.deployMode,client)
[2024-10-21T22:08:46.763+0000] {spark_submit.py:649} INFO - (spark.submit.pyFiles,)
[2024-10-21T22:08:46.763+0000] {spark_submit.py:649} INFO - (spark.ui.port,4050)
[2024-10-21T22:08:46.763+0000] {spark_submit.py:649} INFO - Classpath elements:
[2024-10-21T22:08:46.763+0000] {spark_submit.py:649} INFO - 
[2024-10-21T22:08:46.764+0000] {spark_submit.py:649} INFO - 
[2024-10-21T22:08:46.764+0000] {spark_submit.py:649} INFO - 
[2024-10-21T22:08:47.522+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO SparkContext: Running Spark version 3.5.3
[2024-10-21T22:08:47.523+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO SparkContext: OS info Linux, 6.6.22-linuxkit, aarch64
[2024-10-21T22:08:47.526+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO SparkContext: Java version 17.0.13
[2024-10-21T22:08:47.556+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-10-21T22:08:47.607+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO ResourceUtils: ==============================================================
[2024-10-21T22:08:47.608+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO ResourceUtils: No custom resources configured for spark.driver.
[2024-10-21T22:08:47.608+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO ResourceUtils: ==============================================================
[2024-10-21T22:08:47.609+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO SparkContext: Submitted application: LDA News Topic Modeling
[2024-10-21T22:08:47.618+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memoryOverhead -> name: memoryOverhead, amount: 1024, script: , vendor: , cores -> name: cores, amount: 2, script: , vendor: , memory -> name: memory, amount: 5120, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2024-10-21T22:08:47.623+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO ResourceProfile: Limiting resource is cpus at 2 tasks per executor
[2024-10-21T22:08:47.624+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2024-10-21T22:08:47.650+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO SecurityManager: Changing view acls to: ***
[2024-10-21T22:08:47.652+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO SecurityManager: Changing modify acls to: ***
[2024-10-21T22:08:47.652+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO SecurityManager: Changing view acls groups to:
[2024-10-21T22:08:47.653+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO SecurityManager: Changing modify acls groups to:
[2024-10-21T22:08:47.653+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2024-10-21T22:08:47.764+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO Utils: Successfully started service 'sparkDriver' on port 43715.
[2024-10-21T22:08:47.792+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO SparkEnv: Registering MapOutputTracker
[2024-10-21T22:08:47.816+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO SparkEnv: Registering BlockManagerMaster
[2024-10-21T22:08:47.827+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2024-10-21T22:08:47.831+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2024-10-21T22:08:47.834+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2024-10-21T22:08:47.841+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-c3a84e45-5545-4225-9d67-d7f82e089f9d
[2024-10-21T22:08:47.852+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO MemoryStore: MemoryStore started with capacity 2.2 GiB
[2024-10-21T22:08:47.876+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO SparkEnv: Registering OutputCommitCoordinator
[2024-10-21T22:08:47.991+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:47 INFO JettyUtils: Start Jetty 0.0.0.0:4050 for SparkUI
[2024-10-21T22:08:48.029+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 WARN Utils: Service 'SparkUI' could not bind on port 4050. Attempting port 4051.
[2024-10-21T22:08:48.039+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO Utils: Successfully started service 'SparkUI' on port 4051.
[2024-10-21T22:08:48.102+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2024-10-21T22:08:48.124+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO TransportClientFactory: Successfully created connection to spark-master/172.19.0.4:7077 after 12 ms (0 ms spent in bootstraps)
[2024-10-21T22:08:48.183+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20241021220848-0001
[2024-10-21T22:08:48.187+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 39203.
[2024-10-21T22:08:48.188+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO NettyBlockTransferService: Server created on 17bb0edcf65f:39203
[2024-10-21T22:08:48.188+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2024-10-21T22:08:48.191+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 17bb0edcf65f, 39203, None)
[2024-10-21T22:08:48.192+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO BlockManagerMasterEndpoint: Registering block manager 17bb0edcf65f:39203 with 2.2 GiB RAM, BlockManagerId(driver, 17bb0edcf65f, 39203, None)
[2024-10-21T22:08:48.193+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 17bb0edcf65f, 39203, None)
[2024-10-21T22:08:48.194+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 17bb0edcf65f, 39203, None)
[2024-10-21T22:08:48.289+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO Utils: Using initial executors = 3, max of spark.dynamicAllocation.initialExecutors, spark.dynamicAllocation.minExecutors and spark.executor.instances
[2024-10-21T22:08:48.290+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO ExecutorAllocationManager: Dynamic allocation is enabled without a shuffle service.
[2024-10-21T22:08:48.316+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20241021220848-0001/0 on worker-20241021220759-172.19.0.6-44775 (172.19.0.6:44775) with 2 core(s)
[2024-10-21T22:08:48.317+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO StandaloneSchedulerBackend: Granted executor ID app-20241021220848-0001/0 on hostPort 172.19.0.6:44775 with 2 core(s), 5.0 GiB RAM
[2024-10-21T22:08:48.352+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20241021220848-0001/0 is now RUNNING
[2024-10-21T22:08:48.369+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2024-10-21T22:08:48.464+0000] {spark_submit.py:649} INFO - INFO:__main__:Spark session initialized.
[2024-10-21T22:08:48.464+0000] {spark_submit.py:649} INFO - INFO:__main__:Reading articles from: /opt/***/data/news_articles.json
[2024-10-21T22:08:48.511+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2024-10-21T22:08:48.523+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2024-10-21T22:08:48.887+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO InMemoryFileIndex: It took 17 ms to list leaf files for 1 paths.
[2024-10-21T22:08:48.912+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:48 INFO InMemoryFileIndex: It took 1 ms to list leaf files for 1 paths.
[2024-10-21T22:08:49.495+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:49 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Registered executor NettyRpcEndpointRef(spark-client://Executor) (172.19.0.6:45750) with ID 0,  ResourceProfileId 0
[2024-10-21T22:08:49.498+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:49 INFO ExecutorMonitor: New executor 0 has registered (new total is 1)
[2024-10-21T22:08:49.523+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:49 INFO BlockManagerMasterEndpoint: Registering block manager 172.19.0.6:40555 with 2.8 GiB RAM, BlockManagerId(0, 172.19.0.6, 40555, None)
[2024-10-21T22:08:49.749+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:49 INFO FileSourceStrategy: Pushed Filters:
[2024-10-21T22:08:49.750+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:49 INFO FileSourceStrategy: Post-Scan Filters:
[2024-10-21T22:08:49.846+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:49 INFO MemoryStore: Block broadcast_0 stored as values in memory (estimated size 201.4 KiB, free 2.2 GiB)
[2024-10-21T22:08:49.942+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:49 INFO MemoryStore: Block broadcast_0_piece0 stored as bytes in memory (estimated size 34.7 KiB, free 2.2 GiB)
[2024-10-21T22:08:49.943+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:49 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 17bb0edcf65f:39203 (size: 34.7 KiB, free: 2.2 GiB)
[2024-10-21T22:08:49.945+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:49 INFO SparkContext: Created broadcast 0 from json at NativeMethodAccessorImpl.java:0
[2024-10-21T22:08:49.950+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:49 INFO FileSourceScanExec: Planning scan with bin packing, max size: 4194304 bytes, open cost is considered as scanning 4194304 bytes.
[2024-10-21T22:08:50.009+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:50 INFO SparkContext: Starting job: json at NativeMethodAccessorImpl.java:0
[2024-10-21T22:08:50.018+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:50 INFO DAGScheduler: Got job 0 (json at NativeMethodAccessorImpl.java:0) with 1 output partitions
[2024-10-21T22:08:50.018+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:50 INFO DAGScheduler: Final stage: ResultStage 0 (json at NativeMethodAccessorImpl.java:0)
[2024-10-21T22:08:50.019+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:50 INFO DAGScheduler: Parents of final stage: List()
[2024-10-21T22:08:50.019+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:50 INFO DAGScheduler: Missing parents: List()
[2024-10-21T22:08:50.020+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:50 INFO DAGScheduler: Submitting ResultStage 0 (MapPartitionsRDD[3] at json at NativeMethodAccessorImpl.java:0), which has no missing parents
[2024-10-21T22:08:50.050+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:50 INFO MemoryStore: Block broadcast_1 stored as values in memory (estimated size 16.6 KiB, free 2.2 GiB)
[2024-10-21T22:08:50.051+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:50 INFO MemoryStore: Block broadcast_1_piece0 stored as bytes in memory (estimated size 7.8 KiB, free 2.2 GiB)
[2024-10-21T22:08:50.051+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:50 INFO BlockManagerInfo: Added broadcast_1_piece0 in memory on 17bb0edcf65f:39203 (size: 7.8 KiB, free: 2.2 GiB)
[2024-10-21T22:08:50.052+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:50 INFO SparkContext: Created broadcast 1 from broadcast at DAGScheduler.scala:1585
[2024-10-21T22:08:50.059+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:50 INFO DAGScheduler: Submitting 1 missing tasks from ResultStage 0 (MapPartitionsRDD[3] at json at NativeMethodAccessorImpl.java:0) (first 15 tasks are for partitions Vector(0))
[2024-10-21T22:08:50.059+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:50 INFO TaskSchedulerImpl: Adding task set 0.0 with 1 tasks resource profile 0
[2024-10-21T22:08:50.069+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:50 INFO TaskSetManager: Starting task 0.0 in stage 0.0 (TID 0) (172.19.0.6, executor 0, partition 0, PROCESS_LOCAL, 9609 bytes)
[2024-10-21T22:08:55.012+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:55 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20241021220848-0001/1 on worker-20241021220759-172.19.0.6-44775 (172.19.0.6:44775) with 2 core(s)
[2024-10-21T22:08:55.015+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:55 INFO StandaloneSchedulerBackend: Granted executor ID app-20241021220848-0001/1 on hostPort 172.19.0.6:44775 with 2 core(s), 5.0 GiB RAM
[2024-10-21T22:08:55.017+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:55 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20241021220848-0001/2 on worker-20241021220759-172.19.0.6-44775 (172.19.0.6:44775) with 2 core(s)
[2024-10-21T22:08:55.020+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:55 INFO StandaloneSchedulerBackend: Granted executor ID app-20241021220848-0001/2 on hostPort 172.19.0.6:44775 with 2 core(s), 5.0 GiB RAM
[2024-10-21T22:08:55.057+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:55 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20241021220848-0001/2 is now RUNNING
[2024-10-21T22:08:55.059+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:55 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20241021220848-0001/1 is now RUNNING
[2024-10-21T22:08:56.137+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:56 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Registered executor NettyRpcEndpointRef(spark-client://Executor) (172.19.0.6:58358) with ID 1,  ResourceProfileId 0
[2024-10-21T22:08:56.140+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:56 INFO ExecutorMonitor: New executor 1 has registered (new total is 2)
[2024-10-21T22:08:56.140+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:56 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Registered executor NettyRpcEndpointRef(spark-client://Executor) (172.19.0.6:58362) with ID 2,  ResourceProfileId 0
[2024-10-21T22:08:56.140+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:56 INFO ExecutorMonitor: New executor 2 has registered (new total is 3)
[2024-10-21T22:08:56.165+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:56 INFO BlockManagerMasterEndpoint: Registering block manager 172.19.0.6:36905 with 2.8 GiB RAM, BlockManagerId(1, 172.19.0.6, 36905, None)
[2024-10-21T22:08:56.166+0000] {spark_submit.py:649} INFO - 24/10/21 22:08:56 INFO BlockManagerMasterEndpoint: Registering block manager 172.19.0.6:39161 with 2.8 GiB RAM, BlockManagerId(2, 172.19.0.6, 39161, None)
[2024-10-21T22:11:27.290+0000] {local_task_job_runner.py:346} WARNING - State of this instance has been externally set to None. Terminating instance.
[2024-10-21T22:11:27.291+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2024-10-21T22:11:27.292+0000] {process_utils.py:132} INFO - Sending 15 to group 241. PIDs of all processes in the group: [243, 355, 241]
[2024-10-21T22:11:27.292+0000] {process_utils.py:87} INFO - Sending the signal 15 to group 241
[2024-10-21T22:11:27.293+0000] {taskinstance.py:3092} ERROR - Received SIGTERM. Terminating subprocesses.
[2024-10-21T22:11:27.293+0000] {spark_submit.py:775} INFO - Sending kill signal to spark-submit
[2024-10-21T22:11:27.301+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 762, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 563, in submit
    self._process_spark_submit_log(iter(self._submit_sp.stdout))  # type: ignore
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 612, in _process_spark_submit_log
    for line in itr:
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3094, in signal_handler
    raise AirflowTaskTerminated("Task received SIGTERM signal")
airflow.exceptions.AirflowTaskTerminated: Task received SIGTERM signal
[2024-10-21T22:11:27.304+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=newsapi_etl_with_spark, task_id=spark_submit_task, run_id=manual__2024-10-21T22:08:38.616113+00:00, execution_date=20241021T220838, start_date=20241021T220845, end_date=20241021T221127
[2024-10-21T22:11:27.323+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-10-21T22:11:27.327+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 33 for task spark_submit_task ((pymysql.err.IntegrityError) (1452, 'Cannot add or update a child row: a foreign key constraint fails (`***_db`.`task_fail`, CONSTRAINT `task_fail_ti_fkey` FOREIGN KEY (`dag_id`, `task_id`, `run_id`, `map_index`) REFERENCES `task_instance` (`dag_id`, `task_id`, `run_id`, `map_index`) ON DEL)')
[SQL: INSERT INTO task_fail (task_id, dag_id, run_id, map_index, start_date, end_date, duration) VALUES (%(task_id)s, %(dag_id)s, %(run_id)s, %(map_index)s, %(start_date)s, %(end_date)s, %(duration)s)]
[parameters: {'task_id': 'spark_submit_task', 'dag_id': 'newsapi_etl_with_spark', 'run_id': 'manual__2024-10-21T22:08:38.616113+00:00', 'map_index': -1, 'start_date': datetime.datetime(2024, 10, 21, 22, 8, 45, 934780), 'end_date': datetime.datetime(2024, 10, 21, 22, 11, 27, 303174), 'duration': 161}]
(Background on this error at: https://sqlalche.me/e/14/gkpj); 241)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 762, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 563, in submit
    self._process_spark_submit_log(iter(self._submit_sp.stdout))  # type: ignore
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 612, in _process_spark_submit_log
    for line in itr:
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3094, in signal_handler
    raise AirflowTaskTerminated("Task received SIGTERM signal")
airflow.exceptions.AirflowTaskTerminated: Task received SIGTERM signal

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1910, in _execute_context
    self.dialect.do_execute(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
  File "/home/airflow/.local/lib/python3.8/site-packages/pymysql/cursors.py", line 153, in execute
    result = self._query(query)
  File "/home/airflow/.local/lib/python3.8/site-packages/pymysql/cursors.py", line 322, in _query
    conn.query(q)
  File "/home/airflow/.local/lib/python3.8/site-packages/pymysql/connections.py", line 563, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/home/airflow/.local/lib/python3.8/site-packages/pymysql/connections.py", line 825, in _read_query_result
    result.read()
  File "/home/airflow/.local/lib/python3.8/site-packages/pymysql/connections.py", line 1199, in read
    first_packet = self.connection._read_packet()
  File "/home/airflow/.local/lib/python3.8/site-packages/pymysql/connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "/home/airflow/.local/lib/python3.8/site-packages/pymysql/protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "/home/airflow/.local/lib/python3.8/site-packages/pymysql/err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.IntegrityError: (1452, 'Cannot add or update a child row: a foreign key constraint fails (`***_db`.`task_fail`, CONSTRAINT `task_fail_ti_fkey` FOREIGN KEY (`dag_id`, `task_id`, `run_id`, `map_index`) REFERENCES `task_instance` (`dag_id`, `task_id`, `run_id`, `map_index`) ON DEL)')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 327, in _run_raw_task
    ti.handle_failure(e, test_mode, context, session=session)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3427, in handle_failure
    _handle_failure(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/api_internal/internal_api_call.py", line 139, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1243, in _handle_failure
    TaskInstance.save_to_db(failure_context["ti"], session)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/api_internal/internal_api_call.py", line 139, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3399, in save_to_db
    session.flush()
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 3449, in flush
    self._flush(objects)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 3589, in _flush
    transaction.rollback(_capture_exception=True)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    compat.raise_(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 3549, in _flush
    flush_context.execute()
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/orm/unitofwork.py", line 456, in execute
    rec.execute(self)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/orm/unitofwork.py", line 630, in execute
    util.preloaded.orm_persistence.save_obj(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/orm/persistence.py", line 245, in save_obj
    _emit_insert_statements(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/orm/persistence.py", line 1238, in _emit_insert_statements
    result = connection._execute_20(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1710, in _execute_20
    return meth(self, args_10style, kwargs_10style, execution_options)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 334, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1577, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1953, in _execute_context
    self._handle_dbapi_exception(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 2134, in _handle_dbapi_exception
    util.raise_(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1910, in _execute_context
    self.dialect.do_execute(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
  File "/home/airflow/.local/lib/python3.8/site-packages/pymysql/cursors.py", line 153, in execute
    result = self._query(query)
  File "/home/airflow/.local/lib/python3.8/site-packages/pymysql/cursors.py", line 322, in _query
    conn.query(q)
  File "/home/airflow/.local/lib/python3.8/site-packages/pymysql/connections.py", line 563, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/home/airflow/.local/lib/python3.8/site-packages/pymysql/connections.py", line 825, in _read_query_result
    result.read()
  File "/home/airflow/.local/lib/python3.8/site-packages/pymysql/connections.py", line 1199, in read
    first_packet = self.connection._read_packet()
  File "/home/airflow/.local/lib/python3.8/site-packages/pymysql/connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "/home/airflow/.local/lib/python3.8/site-packages/pymysql/protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "/home/airflow/.local/lib/python3.8/site-packages/pymysql/err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
sqlalchemy.exc.IntegrityError: (pymysql.err.IntegrityError) (1452, 'Cannot add or update a child row: a foreign key constraint fails (`***_db`.`task_fail`, CONSTRAINT `task_fail_ti_fkey` FOREIGN KEY (`dag_id`, `task_id`, `run_id`, `map_index`) REFERENCES `task_instance` (`dag_id`, `task_id`, `run_id`, `map_index`) ON DEL)')
[SQL: INSERT INTO task_fail (task_id, dag_id, run_id, map_index, start_date, end_date, duration) VALUES (%(task_id)s, %(dag_id)s, %(run_id)s, %(map_index)s, %(start_date)s, %(end_date)s, %(duration)s)]
[parameters: {'task_id': 'spark_submit_task', 'dag_id': 'newsapi_etl_with_spark', 'run_id': 'manual__2024-10-21T22:08:38.616113+00:00', 'map_index': -1, 'start_date': datetime.datetime(2024, 10, 21, 22, 8, 45, 934780), 'end_date': datetime.datetime(2024, 10, 21, 22, 11, 27, 303174), 'duration': 161}]
(Background on this error at: https://sqlalche.me/e/14/gkpj)
[2024-10-21T22:11:27.352+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=243, status='terminated', started='22:08:45') (243) terminated with exit code None
[2024-10-21T22:11:27.353+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=241, status='terminated', exitcode=1, started='22:08:45') (241) terminated with exit code 1
[2024-10-21T22:11:27.353+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=355, status='terminated', started='22:08:45') (355) terminated with exit code None
